examples/tv-casting-app/linux/main.cpp

```diff
@@ -237,23 +227,280 @@ void OnContentLauncherFailureResponse(void * context, CHIP_ERROR error)
+
+    void Initialize(NodeId nodeId, chip::FabricIndex fabricIndex)
+    {
+        mNodeId      = nodeId;
+        mFabricIndex = fabricIndex;
+        for (auto & endpointInfo : mEndpoints)
+        {
+            endpointInfo.Reset();
+        }
+        mInitialized = true;
+    }
+
+    NodeId GetNodeId() { return mNodeId; }
+    chip::FabricIndex GetFabricIndex() { return mFabricIndex; }
+
+    TargetEndpointInfo * GetAddEndpoint(EndpointId endpointId)
+    {
+        if (!mInitialized)
+        {
+            return nullptr;
+        }
+        TargetEndpointInfo * endpoint = GetEndpoint(endpointId);
+        if (endpoint != nullptr)
+        {
+            return endpoint;
+        }
+        for (auto & endpointInfo : mEndpoints)
+        {
+            if (!endpointInfo.IsInitialized())
+            {
+                endpointInfo.Initialize(endpointId);
+                return &endpointInfo;
+            }
+        }
+        return nullptr;
+    }
+
+    TargetEndpointInfo * GetEndpoint(EndpointId endpointId)
+    {
+        if (!mInitialized)
+        {
+            return nullptr;
+        }
+        for (auto & endpointInfo : mEndpoints)
+        {
+            if (endpointInfo.IsInitialized() && endpointInfo.GetEndpointId() == endpointId)
+            {
+                return &endpointInfo;
+            }
+        }
+        return nullptr;
+    }
+
+    bool HasEndpoint(EndpointId endpointId)
+    {
+        if (!mInitialized)
+        {
+            return false;
+        }
+        for (auto & endpointInfo : mEndpoints)
+        {
+            if (endpointInfo.IsInitialized() && endpointInfo.GetEndpointId() == endpointId)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    void PrintInfo()
+    {
+        ChipLogProgress(NotSpecified, " TargetVideoPlayerInfo nodeId=0x" ChipLogFormatX64 " fabric index=%d",
+                        ChipLogValueX64(mNodeId), mFabricIndex);
+        for (auto & endpointInfo : mEndpoints)
+        {
+            if (endpointInfo.IsInitialized())
+            {
+                endpointInfo.PrintInfo();
+            }
+        }
+    }
+
+private:
+    static constexpr size_t kMaxNumberOfEndpoints = 5;
+    TargetEndpointInfo mEndpoints[kMaxNumberOfEndpoints];
+    NodeId mNodeId;
+    chip::FabricIndex mFabricIndex;
+    bool mInitialized = false;
+};
+TargetVideoPlayerInfo gTargetVideoPlayerInfo;
+
+void OnDescriptorReadSuccessResponse(void * context, const chip::app::DataModel::DecodableList<chip::ClusterId> & responseList)
+{
+    TargetEndpointInfo * endpointInfo = static_cast<TargetEndpointInfo *>(context);
+
+    ChipLogProgress(AppServer, "Descriptor: Default Success Response endpoint=%d", endpointInfo->GetEndpointId());
+
+    auto iter = responseList.begin();
+    while (iter.Next())
+    {
+        auto & clusterId = iter.GetValue();
+        endpointInfo->AddCluster(clusterId);
+    }
+    gTargetVideoPlayerInfo.PrintInfo();
```

Comment:
Should this log if the iter state is not success here (i.e. decoding of the list failed somewhere)?

Reply:
added comment to clarify